pkg/common/checkeractionwatchdog.go

```diff
@@ -39,9 +40,11 @@ func (f ActionFunc) Do(ctx context.Context) {
 // Watchdog manages a goroutine that can be started and stopped.
 type Watchdog interface {
 	// Start starts a goroutine using the given context.
-	Start(ctx context.Context)
+	Start(context.Context, uint, *backoff.ExponentialBackoff)
 	// Stop stops the goroutine started by Start.
 	Stop()
+	// Confirm gives the confirmation about checkerActionWatchdog check.
+	Confirm(context.Context, uint, *backoff.ExponentialBackoff) bool
```

Comment:
Do we really need to expose this function? It can also be a private function in the `Watchdog` implementation, right?

Reply:
Actually I have also added the unit tests  which required to call `Confirm` from different pkg, that’s why it’s public.
check [here](https://github.com/gardener/etcd-backup-restore/pull/444/files?diff=unified&w=0#diff-ca1dc33ca57f024b5c11aa85848011b962f18bfb602046f3786f6b25676e7261R123)